tests/Api/Admin/OrdersTest.php

```diff
@@ -65,4 +68,224 @@ public function it_gets_orders_for_customer(): void
+            'admin/order/get_shipping_address_of_placed_order_response',
+            Response::HTTP_OK,
+        );
+    }
+
+    /** @test */
+    public function it_updated_a_shipping_address_of_placed_order(): void
+    {
+        $fixtures = $this->loadFixturesFromFiles([
+            'authentication/api_administrator.yaml',
+            'channel.yaml',
+            'order/customer.yaml',
+            'order/order.yaml',
+        ]);
+
+        /** @var AddressInterface $shippingAddress */
+        $shippingAddress = $fixtures['shipping_address'];
+
+        $this->client->request(
+            method: 'PUT',
+            uri: '/api/v2/admin/addresses/' . $shippingAddress->getId(),
+            server: $this->createUpdateRequestHeaders('api@example.com'),
+            content: json_encode([
+                'firstName' => 'Updated: Julia',
+                'lastName' => 'Updated: Kowalska',
+                'company' => 'Updated: Błysk',
+                'street' => 'Updated: Marszałkowska 10',
+                'countryCode' => 'Updated: PL',
+                'city' => 'Updated: Warszawa',
+                'postcode' => 'Updated: 00-001',
+                'phoneNumber' => 'Updated: 48222333444',
+                'provinceCode' => 'Updated: PL-MA',
+                'provinceName' => 'Updated: mazowieckie'
+            ]),
+        );
+
+        $this->assertResponse(
+            $this->client->getResponse(),
+            'admin/order/put_shipping_address_of_placed_order_response',
+            Response::HTTP_OK,
+        );
+    }
+
+    /** @test */
+    public function it_prevents_customer_update_in_shipping_address_of_placed_order(): void
+    {
+        $fixtures = $this->loadFixturesFromFiles([
+            'authentication/api_administrator.yaml',
+            'channel.yaml',
+            'order/customer.yaml',
+            'order/order.yaml',
+        ]);
+
+        /** @var AddressInterface $shippingAddress */
+        $shippingAddress = $fixtures['shipping_address'];
+
+        /** @var CustomerInterface $customerTony */
+        $customerTony = $fixtures['customer_tony'];
+
+        /** @var CustomerInterface $customerDave */
+        $customerDave = $fixtures['customer_dave'];
+
+        $this->client->request(
+            method: 'PUT',
+            uri: '/api/v2/admin/addresses/' . $shippingAddress->getId(),
+            server: $this->createUpdateRequestHeaders('api@example.com'),
+            content: json_encode([
+                'customer' => '/api/v2/admin/customers/' . $customerDave->getId(),
+            ]),
+        );
+
+        $content = $this->client->getResponse()->getContent();
+        Assert::assertNotFalse($content);
+
+        $this->assertResponseCode($this->client->getResponse(), Response::HTTP_OK);
+        $this->assertSame('/api/v2/admin/customers/' . $customerTony->getId(), json_decode($content)->customer);
+    }
+
+    private function createUpdateRequestHeaders(string $adminEmail): array
```

Comment:
Headers for `createUpdateRequestHeaders` and `createFetchRequestHeaders` looks identical, can they be replaced with single method?

For example,
```php
private function buildHeaders(string $adminEmail): array
```

"Fetch"/"update" detail in the name doesn't look important.

Reply:
I'm pretty sure that I started with the PATCH header, and then, when it turned out that I need a PUT method, I overlooked this duplication, good catch